src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -69,17 +70,20 @@ public class PrometheusMetricsReporterConfig extends AbstractConfig {
     private final Listener listener;
     private final boolean listenerEnabled;
     private final Pattern allowlist;
+    private final PrometheusRegistry registry;
 
     /**
      * Constructor.
      *
      * @param props the configuration properties.
+     * @param registry the metrics registry
      */
-    public PrometheusMetricsReporterConfig(Map<?, ?> props) {
+    public PrometheusMetricsReporterConfig(Map<?, ?> props, PrometheusRegistry registry) {
```

Comment:
It's not ideal to have a `registry` field here. But it's required when start the HTTP server.